// js-omdb-api/dist/client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var config_1 = require("./config/config");
require("whatwg-fetch");
var paramsMap = {
    search: "s",
    title: "t",
    imdbId: "i",
    year: "y",
    page: "page",
    season: "Season",
    episode: "Episode",
    plot: "plot",
    type: "type",
    tomatoes: "tomatoes",
};
var createUrlFromParams = function (params) {
    var keys = Object.keys(params);
    var url = config_1.settings.url + "?r=json";
    for (var i = 0; i < keys.length; i++) {
        var key = keys[i];
        var convertedKey = paramsMap[key];
        if (!convertedKey) {
            throw new Error("Invalid key " + key);
        }
        url += "&" + convertedKey + "=" + params[key];
    }
    return url;
};
exports.get = function (params) {
    var url = createUrlFromParams(params);
    return fetch(url)
        .then(function (res) {
        return res.json();
    })
        .then(function (searchResult) {
        if (searchResult.Response === "False") {
            throw Error(searchResult.Error);
        }
        return searchResult;
    });
};
